### PR TITLE
Fix bugs in examples

### PR DESCRIPTION
**Pull request by tkoskela**
I tried building the examples and found and fixed a few bugs

### CODE REVIEW

File: cpp/example/mem_w_algos.cc [hunk 1]
```diff
@@ -58,7 +58,7 @@ int main(int nargs, char const **args) {
   std::vector<t_int> image_index;
   std::tie(uv_data, image_index, w_stacks) = utilities::w_stacking_with_all_to_all(
       uv_data, du, Ju, Jw, world, 100, 0., cost, k_means_rel_diff);
-  for (t_real k = 0; k < uv_data.size(); k++) {
+  for (t_int k = 0; k < uv_data.size(); k++) {
```
Comment (tkoskela) [inline cpp/example/mem_w_algos.cc:61]: Honorary mention to this :bug:

File: cpp/example/padmm_real_data.cc [hunk 2]
```diff
@@ -46,7 +46,7 @@ void padmm(const std::string &name, const t_uint &imsizex, const t_uint &imsizey
   auto const Psi = sopt::linear_transform<t_complex>(sara, imsizey, imsizex);
   const Vector<> dimage = (measurements_transform->adjoint() * uv_data.vis).real();
   Matrix<t_complex> point = Matrix<t_complex>::Zero(imsizey, imsizex);
-  point(std::floor(imsizey / 2), std::floor(imsizex / 2)) = 1.;
+  point(static_cast<int>(std::floor(imsizey / 2)), static_cast<int>(std::floor(imsizex / 2))) = 1.;
```
Comment (20DM) [inline cpp/example/padmm_real_data.cc]: Why not simplify
```
point(static_cast<int>(std::floor(imsizey / 2)), static_cast<int>(std::floor(imsizex / 2)))
```
to
```
point(int(imsizey / 2), int(imsizex / 2))
```
? 🤔

**Review by 20DM — COMMENTED** [review]
See above

File: cpp/example/padmm_real_data.cc (hunk 2 above)
Comment (tkoskela) [reply cpp/example/padmm_real_data.cc]: Yeah I guess that would work. Does `int` give the same rounding as `floor`? (not that we care that much here)

Comment (mmcleod89) [reply cpp/example/padmm_real_data.cc]: static cast to int should give a truncation, so it would do the same as floor followed by a cast.

**Review by 20DM — APPROVED** [review]
LGTM 🚀

**Pull request merged**